src/zenml/post_execution/pipeline.py

```diff
@@ -89,6 +89,32 @@ def get_run(self, name: str) -> "PipelineRunView":
             )
         return run
 
+    def get_run_for_completed_step(self, step_name: str) -> "PipelineRunView":
+        """This method helps you find out which pipeline run produced
+        the cached artifact of a given step.
+
+        Args:
+            step_name: Name of step at hand
+        Return:
+            None if no run is found that completed the given step,
+             else the original pipeline_run
+        """
+        orig_pipeline_run = None
+
+        for run in reversed(self.runs):
+            try:
+                step = run.get_step(step_name)
+                if step.is_completed:
+                    orig_pipeline_run = run
+                    break
+            except KeyError:
+                pass
+        if not orig_pipeline_run:
+            raise RuntimeError("Pipeline Run for cached step could not "
```

Comment:
Maybe key error?

Reply:
Was contemplating it, but key error only seems reasonable if the step name exists in no run - if the step exist in a run but has never been completed I would not consider that a key error

Reply:
Okay that is a technicality i didnt notice. RunTimeError seems wrong though... Maybe ValueError? But KeyError or more generally `LookupError` seem closest to me

Reply:
Alright LookupError works for me - i have also made the Error message a bit more explicit